biscuit-auth/src/error.rs

```diff
@@ -224,7 +224,12 @@ pub enum RunLimit {
 pub enum LanguageError {
     #[error("datalog parsing error")]
     ParseError(ParseErrors),
-    #[error("facts must not contain unbound parameters")]
+    #[error("datalog parameters must all be bound, provided values must all be used.\nMissing parameters: {missing_parameters:?}\nUnused parameters: {unused_parameters:?}")]
+    Parameters {
+        missing_parameters: Vec<String>,
+        unused_parameters: Vec<String>,
+    },
+    #[error("datalog fragments must not contain unbound parameters")]
```

Comment:
This subsumes `Builder` and `UnkownParameters`,  I think I could remove them

Reply:
actually, this only subsumes `Builder`
`UnknownParameters` is when trying to bind a value to a parameter that is not in the datalog fragment